src/ocm/components/clusters/Clusters.tsx

```diff
@@ -44,7 +47,20 @@ const Clusters: React.FC<ClustersProps> = ({ history }) => {
   const deleteClusterAsync = React.useCallback(
     async (clusterId) => {
       try {
+        const infraEnvId = LocalStorageBackedCache.getItem(clusterId);
```

Comment:
This is not enough. If the infraEnvId is not in the cache then we'll need to fetch it form the API.
There is this scenario where the cluster might have been created in one browser/computer but when the users want to delete it they use a different browser. In this case there's going to be a cache miss for sure.